app/model/GroupUser.php

```
@@ -0,0 +1,85 @@
+    ) {
+        parent::__construct($context);
+        $this->user = $user;
+    }
+
+    /**
+     * @param int $groupId
+     * @param int $userId
+     * @param bool $active
+     * @param int|null $invitedByUserId
+     */
+    public function addUser($groupId, $userId, $active = false, $invitedByUserId = null)
```

PHPMD (2.15.0)

[error] The method addUser has a boolean flag argument $active, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)